draft-bbs-signatures.md

```diff
@@ -746,24 +755,87 @@ Inputs:
 
 Parameters:
 
-- r (REQUIRED), non-negative integer. The prime order of the G_1 and G_2 groups,
-     defined by the ciphersuite.
+- r (REQUIRED), non-negative integer. The prime order of the G1 and G2 groups, defined by the ciphersuite.
+- expand_length (REQUIRED), non-negative integer. The number of bytes to read from the xof function, defined by the ciphersuite.
+- dst (REQUIRED), octet string. Domain separation tag.
 
 Outputs:
 
 - (scalar_1, ..., scalar_n), a list of non-zero scalars mod r.
 
 Procedure:
 
-1. h = xof(msg_octets)
+1. dst_prime = (I2OSP(len(dst), 1) || dst)
+
+2. msg_prime = (I2OSP(len(msg), 8) || msg)
+
+3. h = xof(msg_prime || dst_prime || I2OSP(0, 1) || I2OSP(n, 8))
+
+4. for i in (1, ..., n):
+
+5.     scalar_i = OS2IP(h.read(expand_length)) mod r
+
+6.     if scalar_i is 0, go back to step 3
+
+7. return (scalar_1, ..., scalar_n)
+```
+
+#### hash_to_scalar_xmd
+
+Implementations not wishing to use the more performant `hash_to_scalar_xof` operation, can elect to use `hash_to_scalar_xmd`. The `hash_to_scalar_xmd` is based on the `expand_message_xmd` function defined in Section 5.4 of [@!I-D.irtf-cfrg-hash-to-curve], with the addition of checking if the resulting scalar is 0.
+
+```
+result = hash_to_scalar_xmd(msg_octets, n)
+
+Inputs:
+- msg_octets (REQUIRED), octet string. The message to be hashed.
+- n (REQUIRED), non-negative integer. The number of scalars to output.
+
+Parameters:
+
+- r (REQUIRED), non-negative integer. The prime order of the G1 and G2 groups, defined by the ciphersuite.
+- expand_length (REQUIRED), non-negative integer. The number of bytes required to compute each scalar, defined by the ciphersuite.
+- b_in_bytes (REQUIRED), b / 8 for b the output size of H in bits. For example, for b = 256, b_in_bytes = 32.
```

Comment:
For consistency maybe just say "the output size of H measured in bytes, equal to b / 8".

Should we just define these during the procedure based on `b` and `s`?

Reply:
Agreed. Updated to use `b` and `s` instead. Thanks!